server/channels/app/notification.go

```diff
@@ -518,12 +518,11 @@ func (a *App) SendNotifications(c request.CTX, post *model.Post, team *model.Tea
 		}
 	}
 
-	if len(mentionedUsersList) != 0 {
-		message.Add("mentions", model.ArrayToJSON(mentionedUsersList))
-	}
-
-	if len(notificationsForCRT.Desktop) != 0 {
-		message.Add("followers", model.ArrayToJSON(notificationsForCRT.Desktop))
+	if len(mentionedUsersList) > 0 || len(notificationsForCRT.Desktop) > 0 {
+		message.GetBroadcast().AddHook(AddMentionsAndFollowers, map[string]any{
```

Comment:
Note that the event doesn't actually start with the `mentions` or `followers` fields, so even if the hook doesn't run like in a mixed version cluster which doesn't understand broadcast hooks, it'll still send the safer version of the event to clients